test/e2e/addon/nodemonitoringagent.go

```diff
@@ -0,0 +1,197 @@
+package addon
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/eks"
+	"github.com/go-logr/logr"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	clientgo "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	ik8s "github.com/aws/eks-hybrid/internal/kubernetes"
+	"github.com/aws/eks-hybrid/test/e2e/commands"
+	"github.com/aws/eks-hybrid/test/e2e/kubernetes"
+)
+
+const (
+	nodeMonitoringAgentNamespace = "kube-system"
+	nodeMonitoringAgentName      = "eks-node-monitoring-agent"
+	lockupDaemonSetName          = "kernel-soft-lockup"
+	lockupContainerNamespace     = "default"
+
+	nodeMonitoringAgentWaitTimeout = 1 * time.Minute
+)
+
+type NodeMonitoringAgentTest struct {
+	Cluster       string
+	addon         *Addon
+	K8S           clientgo.Interface
+	EKSClient     *eks.Client
+	K8SConfig     *rest.Config
+	Logger        logr.Logger
+	CommandRunner commands.RemoteCommandRunner
+}
+
+func (n *NodeMonitoringAgentTest) Run(ctx context.Context) error {
+	n.addon = &Addon{
+		Cluster:   n.Cluster,
+		Namespace: nodeMonitoringAgentNamespace,
+		Name:      nodeMonitoringAgentName,
+	}
+
+	if err := n.Create(ctx); err != nil {
+		return err
+	}
+
+	if err := n.Validate(ctx); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (n *NodeMonitoringAgentTest) Create(ctx context.Context) error {
+	if err := n.addon.CreateAddon(ctx, n.EKSClient, n.K8S, n.Logger); err != nil {
+		return err
+	}
+
+	if _, err := ik8s.GetAndWait(ctx, nodeMonitoringAgentWaitTimeout, n.K8S.AppsV1().DaemonSets(n.addon.Namespace), n.addon.Name, func(ds *appsv1.DaemonSet) bool {
+		return ds.Status.DesiredNumberScheduled == ds.Status.NumberReady
+	}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (n *NodeMonitoringAgentTest) runKernelError(ctx context.Context) error {
+	nodes, err := ik8s.ListRetry(ctx, n.K8S.CoreV1().Nodes())
+	if err != nil {
+		return err
+	}
+
+	timesRun := 0
+	for _, node := range nodes.Items {
+		for _, ip := range node.Status.Addresses {
+			if ip.Type != v1.NodeInternalIP {
+				continue
```

Comment:
why not just return error here? you end up returning an error if count doesn't match number of nodes, but because you hide the cause, the error doesn't say why it didn't run for all nodes